judge/views/blog.py

```diff
@@ -101,7 +102,12 @@ def get_comment_page(self):
 
     def get_context_data(self, **kwargs):
         context = super(PostView, self).get_context_data(**kwargs)
-        context['og_image'] = self.object.og_image
+
+        if not self.object.og_image or not self.object.summary:
+            metadata = generate_opengraph('generated-meta-blog:%d' % self.object.id, self.object.content, 'blog')
+        context['meta_description'] = self.object.summary or metadata[0]
```

Comment:
Not sure if `self.object.summary` is truly appropriate and we shouldn't just display `metadata[0]` always. `summary` is what's displayed on the home page, which can be very long.

Reply:
Updated to `self.object.summary or self.object.content`.